yogstation/code/modules/guardian/abilities/major/precision.dm

```diff
@@ -0,0 +1,118 @@
+GLOBAL_LIST_INIT(guardian_precision_speedup, list(
+	1 = -0.5,
+	2 = -0.75,
+	3 = -1,
+	4 = -1.25,
+	5 = -1.5
+)) //doing this for consistency's sake
+
+/mob/living/simple_animal/hostile/guardian/ClickOn(atom/A, params)
+	if(istype(loc, /obj/item/gun/ballistic/revolver/emperor))
+		var/obj/item/gun/ballistic/revolver/emperor/gun = loc
+		var/turf/turf_to_throw_gun = get_ranged_target_turf(gun, get_dir(A, gun), 1)
+		if(gun.process_fire(A, src, TRUE, params) && range > 1) //melee cannot go further if they get dropped on the ground
```

Comment:
> range gives the gun more potential to shoot by itself

This is the only place that the `range` stat is considered and it appears to be able to fire itself just fine without any range? What exactly does this do anyways? Throws the gun at what it fires at? Seems very weird

Reply:
if you have no range, you are bound to your summoner, therefore if you are dropped in the ground with F range you won't be able to shoot yourself out of that range